src/runtime/generator.cpp

```diff
@@ -88,6 +111,7 @@ void generatorEntry(BoxedGenerator* g) {
     // we returned from the body of the generator. next/send/throw will notify the caller
     g->entryExited = true;
     threading::popGenerator();
+    freeGeneratorStack(g);
```

Comment:
I think this works out as long as we always keep at least 1 generator stack around, but it feels risky to call freeGeneratorStack while we're on the stack that we're freeing.  Do you think it works to put this on the other side somewhere? (somewhere in generatorSend I guess)